File: Java/selection_sort.java
public class selection_sort {
    public static void sort(int arr[]){
        for(int i=0; i<arr.length-1; i++){
            int min = i;
            for(int j=i+1; j<arr.length; j++){
                if(arr[j]<arr[min]){
                    min = j;
                }
            }
            if(min != i){
                int temp = arr[i];
                arr[i] = arr[min];
                arr[min] = temp;
            }
        }
    }
    public static void main(String args[]){
        int arr[] = {10, 3, 2, 1, 4, 11};
        sort(arr);
        for(int i=0; i<arr.length; i++){
            System.out.print(arr[i] + ", ");
        }
        System.out.println();
    }
}

File: CPP/selectionSort.cpp
#include <iostream>
using namespace std;

void sort(int arr[], int len) {
    int min, temp;
    for(int i = 0; i < len-1; i++) {
        min = i;
        for(int j = i+1; j < len; j++) {
            if(arr[j] < arr[min]) {
                min = j;
            }
        }
        if(min != i) {
            temp = arr[i];
            arr[i] = arr[min];
            arr[min] = temp;
        }
        for(int i = 0; i < len; i++) {
            cout << arr[i] << "  ";
        }
        cout << endl;
    }
}

int main(){
    int arr[] = {6, 3 ,4 ,2 ,1 ,3};
    sort(arr, 6);

    for(int i = 0; i < 6; i++) {
        cout << arr[i] << "  ";
    }
    cout << endl;
}